src/encoder/encoderflacsettings.cpp

```diff
@@ -0,0 +1,122 @@
+/****************************************************************************
+                   encoder.cpp  - encoder API for mixxx
+                             -------------------
+    copyright            : (C) 2017 by Josep Maria Antolín
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "encoder/encoderflacsettings.h"
+#include "recording/defs_recording.h"
+
+const int EncoderFlacSettings::DEFAULT_QUALITY_VALUE = 5;
+const QString EncoderFlacSettings::BITS_GROUP = "FLAC_BITS";
+const QString EncoderFlacSettings::GROUP_COMPRESSION = "FLAC_COMPRESSION";
+
+ 
+EncoderFlacSettings::EncoderFlacSettings(UserSettingsPointer pConfig)
+{
+    m_pConfig = pConfig;
+
+    QList<QString> names;
+    names.append("16 bits");
+    names.append("24 bits");
+    m_radioList.append(OptionsGroup("Bits", BITS_GROUP, names));
+    
+    m_qualList.append(0);
+    m_qualList.append(1);
+    m_qualList.append(2);
+    m_qualList.append(3);
+    m_qualList.append(4);
+    m_qualList.append(5);
+    m_qualList.append(6);
+    m_qualList.append(7);
+    m_qualList.append(8);
+}
+EncoderFlacSettings::~EncoderFlacSettings()
+{
+    
+}
+
+// Returns the list of compression values supported, to assign them to the slider
+QList<int> EncoderFlacSettings::getCompressionValues() const
+{
+    return m_qualList;
+}
+// Sets the compression level
+void EncoderFlacSettings::setCompression(int compression) {
+    if (m_qualList.contains(compression)) {
+        m_pConfig->set(ConfigKey(RECORDING_PREF_KEY, GROUP_COMPRESSION),
+            ConfigValue(compression));
+    } else {
+        qWarning() << "Received a compression value out of range: " << compression;
```

Comment:
Should this set the compression level to EncoderFlacSettings::DEFAULT_QUALITY_VALUE?